packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp

```diff
@@ -0,0 +1,231 @@
+
+  *: Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  *: Neither the name of the University of Wisconsin - Madison nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \brief DTK_ReferenceHexMesh.cpp
+ * \author Stuart R. Slattery
+ * \brief Reference implementation of a parallel hex mesh for unit testing.
+ */
+//---------------------------------------------------------------------------//
+
+#include "DTK_ReferenceHexMesh.hpp"
+#include "DTK_ReferenceHex.hpp"
+#include "DTK_ReferenceNode.hpp"
+#include "DTK_ReferenceHexLocalMap.hpp"
+#include "DTK_ReferenceHexShapeFunction.hpp"
+#include "DTK_ReferenceHexIntegrationRule.hpp"
+#include <DTK_BasicEntitySet.hpp>
+#include <DTK_DBC.hpp>
+
+namespace DataTransferKit
+{
+namespace UnitTest
+{
+//---------------------------------------------------------------------------//
+// Hex mesh constructor.
+ReferenceHexMesh::ReferenceHexMesh(
+    const Teuchos::RCP<const Teuchos::Comm<int> >& comm,
+    const Teuchos::Array<double>& x_edges,
+    const Teuchos::Array<double>& y_edges,
+    const Teuchos::Array<double>& z_edges)
+{
+    // Get comm parameters.
+    int comm_rank = comm->getRank();
+    int comm_size = comm->getSize();    
+    
+    // Create the partitioning. Z partitioning only.
+    DTK_REQUIRE( x_edges.size() > 1 );
+    DTK_REQUIRE( y_edges.size() > 1 );
+    DTK_REQUIRE( z_edges.size() > comm_size );
+
+    int x_num_nodes = x_edges.size();
+    int y_num_nodes = y_edges.size();
+    int z_num_nodes = z_edges.size();
+    DTK_REMEMBER( int total_nodes = x_num_nodes*y_num_nodes*z_num_nodes );
+    
+    int x_num_cells = x_num_nodes - 1;
+    int y_num_cells = y_num_nodes - 1;
+    int z_num_cells = z_num_nodes - 1;
+    DTK_REMEMBER( int total_cells = x_num_cells*y_num_cells*z_num_cells );    
```

Comment:
Yes - they let you evaluate a statement you might want to use in a later DBC check but would not need in the code for a release build.